pkg/cli/server_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"testing"
+
+	"github.com/abcxyz/pkg/cli"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/testutil"
+)
+
+func TestServerCommand(t *testing.T) {
+	t.Parallel()
+
+	testPrivateKeyString, _ := testGeneratePrivateKey(t)
+
+	ctx := logging.WithLogger(context.Background(), logging.TestLogger(t))
+
+	cases := []struct {
+		name   string
+		args   []string
+		env    map[string]string
+		expErr string
+	}{
+		{
+			name: "success",
+			env: map[string]string{
+				"GITHUB_APP_ID":              "123456",
+				"GITHUB_APP_INSTALLATION_ID": "123456",
+				"GITHUB_APP_PRIVATE_KEY_PEM": testPrivateKeyString,
+			},
+		},
+		{
+			name:   "unexpected args",
+			args:   []string{"foo"},
+			expErr: `unexpected arguments: ["foo"]`,
+		},
+		{
+			name: "invalid_config_missing_github_app_id",
+			env: map[string]string{
+				"GITHUB_APP_INSTALLATION_ID": "123456",
+				"GITHUB_APP_PRIVATE_KEY_PEM": testPrivateKeyString,
+			},
+			expErr: `invalid configuration: GITHUB_APP_ID is empty`,
+		},
+		{
+			name: "invalid_config_missing_github_app_installation_id",
+			env: map[string]string{
+				"GITHUB_APP_ID":              "123456",
+				"GITHUB_APP_PRIVATE_KEY_PEM": testPrivateKeyString,
+			},
+			expErr: `invalid configuration: GITHUB_APP_INSTALLATION_ID is empty`,
+		},
+		{
+			name: "invalid_config_missing_private_key_pem",
+			env: map[string]string{
+				"GITHUB_APP_ID":              "123456",
+				"GITHUB_APP_INSTALLATION_ID": "123456",
+			},
+			expErr: `invalid configuration: GITHUB_APP_PRIVATE_KEY_PEM is empty`,
+		},
```

Comment:
not necessary since you already tested one invalid cfg test cases here and all cases in plugin config.